From 59f7266c560b61d7e1e15e304a92282d3548b1b0 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 25 Jul 2018 01:21:05 -0400
Subject: [PATCH] repair bad rcon.ip settings temporarily

accidently missed mapping change, and we defaulted rcon.ip to the server.properties file path

clean up values for people, drop this patch after like 2 weeks.

diff --git a/src/main/java/net/minecraft/server/RemoteControlListener.java b/src/main/java/net/minecraft/server/RemoteControlListener.java
index c237f239f3..1319b3b916 100644
--- a/src/main/java/net/minecraft/server/RemoteControlListener.java
+++ b/src/main/java/net/minecraft/server/RemoteControlListener.java
@@ -25,6 +25,7 @@ public class RemoteControlListener extends RemoteConnectionThread {
         this.h = iminecraftserver.a("rcon.port", 0);
         this.l = iminecraftserver.a("rcon.password", "");
         this.j = iminecraftserver.a("rcon.ip", ((DedicatedServer) iminecraftserver).getServerIp()); // Paper
+        if (this.j.equals(iminecraftserver.d_())) this.j = ((DedicatedServer) iminecraftserver).getServerIp(); // Paper - temporary - remove this after like 2 weeks - repair bad settings
         this.i = iminecraftserver.f();
         if (0 == this.h) {
             this.h = this.i + 10;
-- 
2.18.0

